grease/src/Grease/Refine.hs

```diff
@@ -219,6 +222,20 @@ combiner = C.Combiner $ \mr1 mr2 -> do
         ProveNoHeuristic errs2 ->
           pure (failed (ProveNoHeuristic (errs1 <> errs2)))
 
+type MacawAssertionCallback sym = sym -> W4.Pred sym -> MSM.MacawError sym -> IO (W4.Pred sym)
+type LLVMMemModelCallback sym = LLCS.CallStack -> Mem.BoolAnn sym -> CLLVM.BadBehavior sym -> IO ()
+
+buildErrMaps :: forall t st fs bldr. (bldr ~ (W4.ExprBuilder t st fs)) => IO (IORef (Map.Map (Nonce t C.BaseBoolType) (ErrorDescription bldr)), LLVMMemModelCallback bldr, MacawAssertionCallback bldr)
```

Comment:
- Could use a Haddock
- You probably don't need to explicitly quantify over all the type variables
- There are superfluous parens in the constraint
- I get nervous once we get up to 3-tuples. Might be worth making a record type that this function returns,  especially since you've already gone through the trouble of naming the types of the callbacks.
- The standard name for what you've called `bldr` is `sym`